testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/jackson2/whitelist/WhiteListPolymorphicTypeValidatorTest.java

```diff
@@ -0,0 +1,144 @@
+package org.jboss.resteasy.test.providers.jackson2.whitelist;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.plugins.providers.jackson.WhiteListPolymorphicTypeValidatorBuilder;
+import org.jboss.resteasy.test.providers.jackson2.whitelist.model.AbstractVehicle;
+import org.jboss.resteasy.test.providers.jackson2.whitelist.model.TestPolymorphicType;
+import org.jboss.resteasy.test.providers.jackson2.whitelist.model.air.Aircraft;
+import org.jboss.resteasy.test.providers.jackson2.whitelist.model.land.Automobile;
+import org.jboss.resteasy.utils.PortProviderUtil;
+import org.jboss.resteasy.utils.TestUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ws.rs.client.ClientBuilder;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Collectors;
+
+
+/**
+ * @tpSubChapter Jackson2 provider
+ * @tpChapter Integration tests
+ * @tpSince RESTEasy 3.6.1.SP8
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class WhiteListPolymorphicTypeValidatorTest {
+
+   protected static final Logger logger = Logger.getLogger(WhiteListPolymorphicTypeValidatorTest.class.getName());
+
+   static ResteasyClient client;
+
+   @Deployment(name = "default")
+   public static Archive<?> deploy() {
+      WebArchive war = TestUtil.prepareArchive(WhiteListPolymorphicTypeValidatorTest.class.getSimpleName());
+      war.addClass(WhiteListPolymorphicTypeValidatorTest.class);
+      return TestUtil.finishContainerPrepare(war, null, JaxRsActivator.class, TestRESTService.class,
+            TestPolymorphicType.class, AbstractVehicle.class, Automobile.class, Aircraft.class);
+   }
+
+   @Before
+   public void init() {
+      client = (ResteasyClient)ClientBuilder.newClient();
+   }
+
+   @After
+   public void after() throws Exception {
+      client.close();
+   }
+
+   private String generateURL(String path) {
+      return PortProviderUtil.generateURL(path, WhiteListPolymorphicTypeValidatorTest.class.getSimpleName());
+   }
+
+   /**
+    * @tpTestDetails Client sends POST request with polymorphic type enabled by configuration of {@link WhiteListPolymorphicTypeValidatorBuilder}.
```

Comment:
I recommend adding a comment here explaining that the build has system properties set that control the behaviour here. Otherwise, it might not be clear by looking at the code why `testGood` would pass.

Reply:
You are right. I've just added an explanation there.